crates/extensions/tedge_flows/src/transformers/mod.rs

```diff
@@ -0,0 +1,159 @@
+use crate::js_value::JsonValue;
+use crate::FlowError;
+use crate::LoadError;
+use crate::Message;
+use std::collections::HashMap;
+use std::time::SystemTime;
+
+mod add_timestamp;
+mod cap_payload_size;
+mod set_topic;
+
+pub trait Transformer: Send + Sync + 'static {
+    fn name(&self) -> &str;
+
+    fn on_message(
+        &self,
+        timestamp: SystemTime,
+        message: &Message,
+        config: &JsonValue,
+    ) -> Result<Vec<Message>, FlowError>;
+
+    fn is_periodic(&self) -> bool {
+        false
+    }
+
+    fn on_interval(
```

Comment:
What's the built-in transformer that you have in mind that could be periodic?

Reply:
I see no reasons to give less power to built-in transformers. Monitoring & limiting message rate is an example of a  generic transformer that would be useful.